pkg/compact/retention.go

```diff
@@ -17,6 +18,8 @@ import (
 	"github.com/thanos-io/thanos/pkg/objstore"
 )
 
+type RetentionPolicy map[ResolutionLevel]time.Duration
```

Comment:
Could we add a comment for this exported type?

Reply:
Will do !